rct_optimizations/src/rct_optimizations/covariance_analysis.cpp

```diff
@@ -171,37 +171,40 @@ CovarianceResult computeCovariance(ceres::Problem &problem,
 }
 
 CovarianceResult computeCovarianceInTangentSpace(ceres::Problem &problem,
-                                                 const std::vector<std::vector<std::string>>& parameter_names,
-                                                 const std::vector<std::vector<int>>& masks,
+                                                 const std::map<double*, std::vector<std::string>>& param_names,
+                                                 const std::map<double*, std::vector<int>>& param_masks,
                                                  const ceres::Covariance::Options& options)
 {
   std::vector<double *> parameter_blocks;
   problem.GetParameterBlocks(&parameter_blocks);
   const std::vector<const double *> param_blocks_const(parameter_blocks.begin(), parameter_blocks.end());
 
   // 0. Check user-specified arguments
-  if (parameter_blocks.size() != parameter_names.size())
+  if (parameter_blocks.size() != param_names.size())
     throw CovarianceException("Provided vector parameter_names is not same length as provided number of parameter blocks");
 
+  if (parameter_blocks.size() != param_masks.size())
+    throw CovarianceException("Provided vector param_masks is not same length as provided number of parameter blocks");
+
   Eigen::Index n_params_in_selected = 0;
   std::vector<Eigen::Index> tagent_space_indices;
   std::vector<std::string> tagent_space_labels;
-  for (std::size_t index_block = 0; index_block < parameter_blocks.size(); index_block++)
+  for (double* b : parameter_blocks)
   {
-    int block_size = problem.ParameterBlockSize(parameter_blocks[index_block]);
-    if (static_cast<std::size_t>(block_size) != parameter_names[index_block].size())
+    int block_size = problem.ParameterBlockSize(b);
+    if (static_cast<std::size_t>(block_size) != param_names.at(b).size())
     {
       std::stringstream ss;
-      ss << "Number of parameter labels provided for block " << index_block << " does not match actual number of parameters in that block: " \
-         << "have " << parameter_names[index_block].size() << " labels and " << block_size << " parameters";
+      ss << "Number of parameter labels provided for block does not match actual number of parameters in that block: " \
+         << "have " << param_names.at(b).size() << " labels and " << block_size << " parameters";
```

Comment:
Add the name of the block in the error message

```suggestion
      ss << "Number of parameter labels provided for block '" << param_names.at(b).first << "' does not match actual number of parameters in that block: " \
         << "have " << param_names.at(b).size() << " labels and " << block_size << " parameters";
```